mindmap/mindmap.py

```diff
@@ -251,11 +284,29 @@ def upload_file(self, data, suffix="") -> None: # pylint: disable=unused-argumen
         )
 
     @XBlock.json_handler
-    def studio_submit(self, data, suffix=""):  # pylint: disable=unused-argument
+    def upload_file_instructor(self, data, _suffix="") -> None:
```

Comment:
where is suffix used here?

Reply:
why are upload_file_instructor / student different? Since we can use _suffix to differentiate instructor or student

Reply:
Questions: 
- why do we have to differentiate between the two saves? 
- is it relevant to separate /students from /instructors mindmaps saves? From the LMS they look the same, don't they?
- Is it useful if we separate those submission files?

Reply:
> Where is suffix used here?
> Why are upload_file_instructor / student different? Since we can use _suffix to differentiate instructor or student

I'm not sure what the `suffix` is for in the handler method, it's already set by default. However, in the `data`, we could include another key that would allow me to know if it is an instructor or student upload, and thus reuse the handler.

> why do we have to differentiate between the two uploads across the implementation?
> Is it relevant to separate /students from /instructors mindmaps saves? From the LMS they look the same, don't they?

Separating the file into a separate folder for the instructor and for each student (according to their anonymous_user_id) allows us to have a unique key/path to which we can refer to the mind map depending on the situation.

> Is it useful if we separate those submission files with the suffix instructor / student for future implementations?

I don't know how useful it would be to create a folder for the students, maybe to keep order in the bucket. Do you have any ideas in mind?